### ISSUE TITLE
more seemless patching

### ISSUE THREAD

**Issue by delanym** [google/error-prone#3350]
When many projects are processed and patched (like in a multi-module build with hundreds of modules), applying the patches is not straight-forward.

The one option is to setup the plugin with 
`<arg>-Xplugin:ErrorProne -XepAllErrorsAsWarnings -XepPatchChecks:DefaultCharset -XepPatchLocation:${project.build.directory}</arg>`

create a script
```shell
#/tmp/patchit
cd "$(dirname $1)/.."
patch -p1 -u -i target/error-prone.patch
```
and run
```
mvn verify -Perrorprone --threads=8
find . -iname error-prone.patch | xargs -I % -P 1 /tmp/patchit %
```
_i couldn't get `patch -d` option working, and note how I have to `xargs -P 1`_

The other option is to create the patch file first as an append only file, and then build, apply, remove:
```shell
touch error-prone.patch
sudo chattr +a error-prone.patch
mvn verify -Perrorprone --threads=1
patch -p0 -u -i error-prone.patch
sudo chattr -a error-prone.patch
rm error-prone.patch
```

I suggestion one or both of the following:

- add an "append to patch file" option to avoid the chattr stuff
- add an option to apply the patch directly on the source files as with git this simply shows up as changes to the working tree and can be applied/rejected

**Comment by Stephan202** [comment]
> add an option to apply the patch directly on the source files as with git this simply shows up as changes to the working tree and can be applied/rejected

This is [supported](https://github.com/google/error-prone/blob/b4d427c93534c7a366b1bd9273aa906bf8e31f9a/check_api/src/main/java/com/google/errorprone/ErrorProneOptions.java#L427) by specifying `-XepPatchLocation:IN_PLACE`.

(It might be nice to add this to the [relevant section](https://errorprone.info/docs/patching) on the website. :bulb:)

**Comment by delanym** [comment]
Its also not clear what rules can be patched. Is it ONLY MissingOverrides and DefaultCharset?

**Comment by delanym** [comment]
@Stephan202 thanks, by the logs it looks like it works, but now I cant get any patches to apply https://github.com/google/error-prone/issues/3349

**Comment by Stephan202** [comment]
> Is it ONLY MissingOverrides and DefaultCharset?

Nope. Most Error Prone checks come with a [suggested fix](https://github.com/google/error-prone/search?q=suggestedfix). In general, if an error or warning message contains a "Did you mean" suggestion, then patching is supported.

**Comment by Stephan202** [comment]
@delanym that could be implemented using a flag. But I'm not an Error Prone maintainer, so I'm not sure such a PR would be accepted. Let's see whether any of the maintainers replies. :eyes:
